azure-ipam/logger/logger.go

```diff
@@ -6,56 +6,48 @@ import (
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
 )
 
 type Config struct {
-	Level            string
-	OutputPaths      string // comma separated list of paths
-	ErrorOutputPaths string // comma separated list of paths
+	Level       string // Debug by default
+	Filename   	string
+	MaxSize    	int    // megabytes
+	MaxBackups 	int    // # of backups, no limitation by default
+	MaxAge     	int    // days, no limitation by default
 }
 
 // NewLogger creates and returns a zap logger and a clean up function
 func New(cfg *Config) (*zap.Logger, func(), error) {
-	loggerCfg := &zap.Config{}
-
-	level, err := zapcore.ParseLevel(cfg.Level)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "failed to parse log level")
-	}
-	loggerCfg.Level = zap.NewAtomicLevelAt(level)
-
-	loggerCfg.Encoding = "json"
-	loggerCfg.OutputPaths = getLogOutputPath(cfg.OutputPaths)
-	loggerCfg.ErrorOutputPaths = getErrOutputPath(cfg.ErrorOutputPaths)
-	loggerCfg.EncoderConfig = zapcore.EncoderConfig{
-		TimeKey:     "time",
-		MessageKey:  "msg",
-		LevelKey:    "level",
-		EncodeLevel: zapcore.LowercaseLevelEncoder,
-		EncodeTime:  zapcore.ISO8601TimeEncoder,
-	}
-
-	logger, err := loggerCfg.Build()
-	if err != nil {
+	//check the filepath is not empty
+	if cfg.Filename == "" {
+		err := errors.New("no Filename is provided")
 		return nil, nil, errors.Wrapf(err, "failed to build zap logger")
 	}
-
+	// set the log level - Debug by default
+	var logLevel zapcore.Level
+	if strings.ToLower(cfg.Level) == "info" {
+		logLevel = zap.InfoLevel
+	} else if strings.ToLower(cfg.Level) == "error" {
+		logLevel = zap.ErrorLevel
+	} else {
+		logLevel = zap.DebugLevel
+	}
+	//define a lumberjack fileWriter
+	logFileWriter := zapcore.AddSync(&lumberjack.Logger{
+		Filename:   cfg.Filename,
+		MaxSize:    cfg.MaxSize, // megabytes
+	})
+	//define the log encoding
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	jsonEncoder := zapcore.NewJSONEncoder(encoderConfig)
+	//create a new zap logger
+	core := zapcore.NewCore(jsonEncoder, logFileWriter, logLevel)
+	logger := zap.New(core)
 	cleanup := func() {
 		_ = logger.Sync()
 	}
-	return logger, cleanup, nil
-}
-
-func getLogOutputPath(paths string) []string {
-	if paths == "" {
-		return nil
-	}
-	return strings.Split(paths, ",")
-}
 
-func getErrOutputPath(paths string) []string {
-	if paths == "" {
-		return nil
-	}
-	return strings.Split(paths, ",")
-}
+	return logger, cleanup, nil
+}
```

Comment:
configure your editor to add trailing newlines on save

Reply:
Will address in the new commit